### ISSUE TITLE
It seems it doesn't have module libopencv_core.so.3.4 on Ubuntu 18.04

### ISSUE THREAD

**Issue by gusbemacbe** [justadudewhohacks/opencv-electron#11]
# Details

* **OS version:** Ubuntu 18.04 64bits
* **Electron version:** 2.0.0
* **electron-build:** 1.7.3
* **Node version:** 9.11.1
* **npm version:** 6.0.0
* **opencv4nodejs:** 4.4.0

1. I forked `electron-quick-start` from https://github.com/electron/electron-quick-start;
2. I renamed it to `electron_opencv`;
3. I ran `npm install --save opencv4nodejs` inside this folder;
4. I readjusted `package.json`:

```json
{
  "name": "electron-opencv",
  "version": "1.0.0",
  "description": "A minimal Electron application with OpenCV",
  "main": "main.js",
  "scripts": {
    "electron-rebuild": "electron-rebuild -w opencv4nodejs",
    "install": "npm run electron-rebuild",
    "start": "electron main.js"
  },
  "repository": "https://github.com/gusbemacbe/electron-opencv",
  "keywords": [
    "Electron",
    "OpenCV"
  ],
  "author": "Gustavo Costa",
  "license": "CC0-1.0",
  "devDependencies": {
    "electron": "^2.0.0",
    "eslint": "^4.19.1",
    "eslint-plugin-react": "^7.7.0"
  },
  "dependencies": {
    "electron": "^2.0.0",
    "electron-rebuild": "^1.7.3",
    "opencv4nodejs": "^4.4.0"
  }
}
```
5. I copied something from your repository subfolder `plain-js` - `app`, `assets` and `main.js`;
6. I ran `npm install` and `npm run electron-rebuild`;
7. Then finally I ran `npm start` and it almost worked. But what didn't work is that after you uploaded an image, it wasn't loaded due to lack of this module.

See the two errors:

```
ELECTRON_ASAR.js:172 Uncaught Error: libopencv_core.so.3.4: cannot open shared object file: No such file or directory
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:671:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
    at Function.Module._load (module.js:496:3)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/ubuntu/Github/electron-opencv/node_modules/opencv4nodejs/lib/opencv4nodejs.js:11:30)
    at Object.<anonymous> (/home/ubuntu/Github/electron-opencv/node_modules/opencv4nodejs/lib/opencv4nodejs.js:18:3)
```
And

```
ncaught TypeError: Cannot read property 'imdecode' of undefined
    at decodeImageFromBase64 (image-helpers.js:7)
    at FileReader.reader.onload (index.html:279)
```

**Comment by gusbemacbe** [comment]
Observe that after adding `const cv = require('opencv4nodejs');` to `main.js`, it also reported the same missed module.

**Comment by gusbemacbe** [comment]
Reference to https://github.com/GaoHongchen/DIPDemo/issues/1.

**Comment by gusbemacbe** [comment]
I found this module on the folder `opencv-build` from `node_modules` at this folder of mine , but I didn't figure out which folder to which I will copy these modules.

**Comment by felipebutcher** [comment]
I am having this same issue when trying to package the application. But for just starting the app it works. OpenCV is built when you do these steps. Make sure it compiled fine. Search for the file libopencv_core.so.3.4 inside your node_modules/opencv-build.

**Comment by justadudewhohacks** [comment]
I am not familar with how to package stuff with electron, but I'll just leave the link to [this](https://github.com/justadudewhohacks/opencv4nodejs/issues/204) issue here, which might be helpful in order to package electron + opencv4nodejs.

**Comment by felipebutcher** [comment]
got the issue solved copying the files from resources/app/node_modules/opencv-build/opencv/build/lib to this one: 
![screenshot from 2018-12-04 23-35-23](https://user-images.githubusercontent.com/4549278/49484332-ca811f80-f81d-11e8-8b64-452c19c1488a.png)

**Comment by mgs96** [comment]
I had the same issue with a nodejs project and this solved it!

**Comment by felipebutcher** [comment]
Hey guys! Got this again now with a Windows packaged release. It's very weird because it works in two machines, and the other one no. In one of the two that work, there's nothing dev related, neither opencv installed. Checked programs installed and the only thing I found that could be related was Microsoft Visual C++ 2013 (x86). 

In other words... both windows machines are very similar and works only in one. Maybe something external?
